adagrams/game.py

```diff
@@ -1,11 +1,170 @@
+import random
+from collections import Counter
+
+LETTER_POOL = {
+  'A': 9,
+  'B': 2,
+  'C': 2,
+  'D': 4,
+  'E': 12,
+  'F': 2,
+  'G': 3,
+  'H': 2,
+  'I': 9,
+  'J': 1,
+  'K': 1,
+  'L': 4,
+  'M': 2,
+  'N': 6,
+  'O': 8,
+  'P': 2,
+  'Q': 1,
+  'R': 6,
+  'S': 4,
+  'T': 6,
+  'U': 4,
+  'V': 2,
+  'W': 2,
+  'X': 1,
+  'Y': 2,
+  'Z': 1
+}
+
+SCORE_CHART = {
+  'A': 1,
+  'B': 3,
+  'C': 3,
+  'D': 2,
+  'E': 1,
+  'F': 4,
+  'G': 2,
+  'H': 4,
+  'I': 1,
+  'J': 8,
+  'K': 5,
+  'L': 1,
+  'M': 3,
+  'N': 1,
+  'O': 1,
+  'P': 3,
+  'Q': 10,
+  'R': 1,
+  'S': 1,
+  'T': 1,
+  'U': 1,
+  'V': 4,
+  'W': 4,
+  'X': 8,
+  'Y': 4,
+  'Z': 10
+}
+
+LETTER_POOL_LIST = []
+temp_list = []
+
 def draw_letters():
-    pass
 
-def uses_available_letters(word, letter_bank):
-    pass
+    # Creates a comprehensive list which includes the total quanity of each available letter from the LETTER_POOL dictionary as individual elements.  
+    for letter, quantity in LETTER_POOL.items():
+        temp_list.append([letter] * quantity)
+ 
+    for sublist in temp_list:
+        for letter in sublist:
+            LETTER_POOL_LIST.append(letter)
+
+    # Chooses 10 random letters as hand, excluding the ones that are already chosen.
+    #hand = random.sample(LETTER_POOL_LIST, 10)
+
+    hand = []
+    while len(hand) < 10:
+        random_letter = random.choice(LETTER_POOL_LIST)
+        random_letter_index = LETTER_POOL_LIST.index(random_letter)
+        LETTER_POOL_LIST.pop(random_letter_index)
+        hand.append(random_letter)
+
+    return hand
+
+draw_letters()
+     
+def uses_available_letters(word, letter_bank):       
+    word_verification = []
+    hand_letter_count = {}
+    word_letter_count = {}
+    count = 0
+    capitalized_word = word.upper()
+
+    for letter in letter_bank:
+        if letter not in hand_letter_count:
+            count = 1
+            hand_letter_count[letter] = count
+        else: 
+            count += 1
+            hand_letter_count[letter] = count
+      
+    for letter in capitalized_word:
+        letter_quantity = capitalized_word.count(letter)
+        word_letter_count[letter] = letter_quantity
+
+        if letter in letter_bank and word_letter_count[letter] <= hand_letter_count[letter]:
+            word_verification.append("True")
+        else:
+            word_verification.append("False")
+      
+    if "False" in word_verification:
+        return False
+    else:
+        return True
+      
 
 def score_word(word):
-    pass
+    word_score = {}
+
+    if word == "":
+       word_score["empty"] = 0
+
+    for letter in word.upper():
+        if letter in SCORE_CHART:
+            if letter not in word_score:
+                word_score[letter] = SCORE_CHART[letter]
+            else:
+                word_score[letter] += SCORE_CHART[letter]
+
+    if 7 <= len(word) <= 10:
+        word_score["bonus"] = 8
+      
+    return sum(word_score.values())
```

Comment:
I think your approach to this problem is interesting and extensible-- I think that adding `"bonus"` key and `sum`ming `word_score.values()` is really smart for an expanded project. In this situation, if you wanted to refactor and make `word_score` an integer, instead of a `{}`, then that would also pass the tests.